modules/storages/app/common/storages/peripherals/storage_interaction/nextcloud/util.rb

```diff
@@ -70,7 +70,9 @@ def origin_user_id(caller:, storage:, auth_strategy:)
               case auth_strategy.key
               when :basic_auth
                 ServiceResult.success(result: storage.username)
-              when :oauth_user_token
+              when :oauth_user_token || :sso_user_token
+                # README: So, this might lead to problems. How can we ensure, that at this point in time,
+                # the remote identity is already created for token exchange?
                 origin_user_id = RemoteIdentity.where(user_id: auth_strategy.user, oauth_client: storage.oauth_client)
```

Comment:
Can't we "simply" turn this into a "find or create" style method for the SSO case?

I.e. if we find a remote identity, we take it. If we don't find one, we make an API call to figure it out and store it.

This means the first time that an SSO user needs a remote identity, they will fetch it via an API call. At all subsequent times, we will have it "cached" in our remote identities.

Reply:
From our IRL discussion: The current plan is that there might an an event triggered from the token exchange or y from the request of a token for a certain audience.

Since those events are handled synchronously, there is no race condition that can happen here. Once we arrive at this point, the event should have created the `RemoteIdentity`.

Essentially, we can keep the code as-is.